text/0000-dot-profile-utility-buildpack.md

```diff
@@ -0,0 +1,161 @@
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+As part of RFC 93, [`.profile` scripts will cease to be supported by the platform API](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md#appprofile).
+This RFC proposes developing a [utility buildpack](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md) to support `.profile` scripts to prevent regressions after RFC 93 is implemented.
+
+# Definitions
+[definitions]: #definitions
+
+*utility buildpack*: A buildpack officially supported by the Buildpack Authors' Tooling Team per [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md) that provides a generic capability.
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md) resolves to remove shell-specific logic from the CNB Specification.
+Part of this includes removing support for `.profile` script in a future version of the Platform API.
+RFC 93 recommends supporting the `.profile` script functionality in a utility buildpack to avoid regressions.
+This proposal is to develop and support the `.profile` utility buildpack, allowing RFC 93 to be implemented without regression.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The target persona is a platform operator or implementor who wants to update to the latest platform API, while maintaining the `.profile` functionality for application developers.
+
+We propose developing and supporting a buildpack to provide an identical interface to the existing `.profile` functionality.
+It will:
+
+- Detect a `.profile` file in the app dir
+- Wrap the `.profile` file so that it implements the `exec.d` interface
+- Add the `exec.d` executable to the `<layer>/exec.d` directory so the launcher will apply it
+
+We propose the ID `buildpacksio/profile` for this buildpack (cf. [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md#what-it-is)).
+
+## Example 1 (environment variables)
+
+Here is an example of a `.profile` script, inspired by paketo-buildpacks/node-engine:
+
+```
+memory_in_bytes="$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)"
+MEMORY_AVAILABLE="$(( $memory_in_bytes / ( 1024 * 1024 ) ))"
+export MEMORY_AVAILABLE
+```
+
+The wrapper should ensure that the `MEMORY_AVAILABLE` environment variable is set in the environment with the proper value.
+
+## Example 2 (file system side effects)
+
+With this `.profile` script:
+
+```
+echo 'hello' >> "$HOME/hello"
+```
+
+The wrapper would not need to set any environment variables, but should maintain the side effect of creating the `$HOME/hello` file.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+We will write two small programs to implement the buildpack.
```

Comment:
There are some minor differences in how the lifecycle executes exec.d scripts vs `.profile` right now. The CWD for the `.profile` script is the process working directory where as it is the app directory for the exec.d scripts. How do we plan on dealing with this edge case?